R/plotTagMatrix.R

```diff
@@ -1716,32 +1782,756 @@ peakHeatmap.internal2 <- function(tagMatrix, xlim, listFlag, color, xlab, ylab,
+                                               upstream=1000, 
+                                               downstream=1000,
+                                               xlab="", 
+                                               ylab="", 
+                                               title=NULL,
+                                               palette=NULL, 
+                                               ncol = NULL,
+                                               nrow = NULL,
+                                               facet_label_text_size = 12){
+
+  binning_Flag <- attr(tagMatrix[[1]],"is.binning")
+  if(binning_Flag) nbin <- dim(tagMatrix[[1]])[2]
+  
+  type <- attr(tagMatrix,"type")
+  body_Flag <- FALSE
+  if(attr(tagMatrix[[1]],"type") == "body"){
+    body_Flag <- TRUE
+    label <- attr(tagMatrix,"label")
+  }
+  
+  name_of_list <- as.list(names(tagMatrix))
+  
+  peak_list <- lapply(name_of_list,function(x){
+    
+    tagMatrix[[x]] <- t(apply(tagMatrix[[x]], 1, function(x) x/max(x)))
+    ii <- order(rowSums(tagMatrix[[x]]))
+    tagMatrix[[x]] <- tagMatrix[[x]][ii,]
+    
+    colnames(tagMatrix[[x]]) <- seq_len(dim(tagMatrix[[x]])[2])
+    rownames(tagMatrix[[x]]) <- seq_len(dim(tagMatrix[[x]])[1])
+    tagMatrix[[x]] <- tagMatrix[[x]] %>% as.data.frame() %>% 
+      rownames_to_column("sample_ID") %>%
+      pivot_longer(-c(sample_ID),names_to = "coordinate", 
+                   values_to = "values")
+    tagMatrix[[x]]$coordinate <- as.numeric(tagMatrix[[x]]$coordinate)
+    tagMatrix[[x]]$sample <- x
+    return(tagMatrix[[x]])
+  })
+  
+  peak_df <- list_to_dataframe(peak_list)
+  
+  sample_ID <- coordinate <- NULL
+  
+  p <- ggplot(peak_df, aes(x = coordinate,y = sample_ID)) + 
+    geom_tile(aes(fill = values)) +
+    scale_fill_distiller(palette = palette)  +
+    theme(axis.text.y=element_blank(),
+          axis.ticks.y=element_blank(),
+          axis.line.y = element_blank(),
+          panel.grid=element_blank(),
+          panel.background = element_blank()) +
+    labs(x = xlab, y = ylab, title = title)
+  
+  if(body_Flag){
+    
+    if(inherits(upstream, 'rel')){
+      
+      p <- p + scale_x_continuous(breaks=c(1, 
+                                           floor(nbin*(as.numeric(upstream)*100/(100+(as.numeric(upstream)+as.numeric(downstream))*100))),
+                                           floor(nbin*((as.numeric(upstream)*100+25)/(100+(as.numeric(upstream)+as.numeric(downstream))*100))),
+                                           floor(nbin*((as.numeric(upstream)*100+50)/(100+(as.numeric(upstream)+as.numeric(downstream))*100))),
+                                           floor(nbin*((as.numeric(upstream)*100+75)/(100+(as.numeric(upstream)+as.numeric(downstream))*100))),
+                                           floor(nbin*((as.numeric(upstream)*100+100)/(100+(as.numeric(upstream)+as.numeric(downstream))*100))),
+                                           nbin),
```

Comment:
are these codes appear in multiple places?
You need to wrap the codes into a function.